api/v1alpha1/etcdpeer_types.go

```diff
@@ -81,6 +94,25 @@ type EtcdPeerList struct {
 	Items           []EtcdPeer `json:"items"`
 }
 
+var _ webhook.Defaulter = &EtcdPeer{}
+
+// Default sets default values for optional EtcdPeer fields.
+// This is used in webhooks and in the Reconciler to ensure that nil pointers
+// have been replaced with concrete pointers.
+// This avoids nil pointer panics later on.
+func (o *EtcdPeer) Default() {
+	if o.Spec.Storage.VolumeClaimTemplate.AccessModes == nil {
+		o.Spec.Storage.VolumeClaimTemplate.AccessModes = []corev1.PersistentVolumeAccessMode{
+			corev1.ReadWriteOnce,
+		}
+	}
+
+	if o.Spec.Storage.VolumeClaimTemplate.VolumeMode == nil {
+		o.Spec.Storage.VolumeClaimTemplate.VolumeMode = &defaultVolumeMode
+	}
+}
+
 func init() {
 	SchemeBuilder.Register(&EtcdPeer{}, &EtcdPeerList{})
+	defaultVolumeMode = corev1.PersistentVolumeFilesystem
```

Comment:
nit: any reason this needs to be assigned in `init()` rather than inline with the definition on ln10?

Reply:
I need a pointer to it, and none of the `pointer` helpers can handle corev1.PersistentVolumeMode type.
It seemed nice to define it globally.

Reply:
Not sure where you're actually using pointers here, so not sure how this applies? 🤔

```golang
var (
    defaultVolumeMode = corev1.PersistentVolumeFilesystem
)
```

And then referencing it as `&defaultVolumeMode` should work?

Reply:
i don't think that stops us doing something like
```
var (
    defaultVolumeMode = corev1.PersistentVolumeFilesystem
)
```
then later on
```
myVar := &defaultVolumeMode
```
but don't let this block you merging, its just a nit

Reply:
Ah, yeah, my mistake. Fixed.